auto-self-nominate/index.ts

```diff
@@ -0,0 +1,179 @@
+            .sendRpcRequest("chain_getTermMetadata", [bestBlockNumber])
+            .then(result => {
+                if (result === null) {
+                    resolve(null);
+                }
+                const [prevTermLastBlockNumber, currentTermId] = result;
+                if (
+                    typeof prevTermLastBlockNumber === "number" &&
+                    typeof currentTermId === "number"
+                ) {
+                    resolve(currentTermId);
+                }
+                reject(
+                    Error(
+                        `Expected getTermMetadata to return [number, number] | null but it returned ${result}`,
+                    ),
+                );
+            })
+            .catch(reject);
+    });
+}
+
+async function sendSelfNominateTransaction(
+    accountInfo: { account: string; passphrase: string },
+    params: { deposit: U64; metadata: string },
+) {
+    const tx = createSelfNomination(params);
+    const { account, passphrase } = accountInfo;
+    await sdk.rpc.account.sendTransaction({
+        tx,
+        account,
+        passphrase,
+    });
+}
+
+function createSelfNomination(params: { deposit: U64; metadata: string }): Custom {
+    const { deposit, metadata } = params;
+    const handlerId = new U64(STAKE_ACTION_HANDLER_ID);
+    const actionTag = ACTION_TAG_SELF_NOMINATE;
+    const bytes = RLP.encode([actionTag, deposit, metadata]);
+    return new Custom(
+        {
+            handlerId,
+            bytes,
+        },
+        networkId(SERVER),
+    );
+}
+
+async function needsNomination(
+    address: string,
+    nominationFrequencyLevel: number,
+): Promise<boolean> {
+    const bestBlockNumber = await sdk.rpc.chain.getBestBlockNumber()!;
+    const candidates: Map<string, number> = await (async () => {
+        const retval = new Map();
+        const data = (await sdk.rpc.engine.getCustomActionData(
+            2,
+            ["Candidates"],
+            bestBlockNumber,
+        ))!;
+        const list: Buffer[][] = RLP.decode(Buffer.from(data, "hex"));
+        list.map(([encodedPubkey, _, encodedNominationEnd]) =>
+            retval.set(
+                decodePlatformAddressFromPubkey(encodedPubkey).value,
+                decodeNumber(encodedNominationEnd),
+            ),
+        );
+        return retval;
+    })();
+    const nominationEndAt = candidates.get(address);
+    if (nominationEndAt == null) {
+        throw new Error(`SelfNominate first with specific deposit vakue before repeating`);
+    } else {
+        const currentTermId = (await getCurrentTermId(bestBlockNumber))!;
+        return nominationEndAt - currentTermId < nominationFrequencyLevel;
+    }
+}
+
+async function main() {
+    const accountAddress = getConfig("ACCOUNT_ADDRESS");
+    const passphrase = getConfig("PASSPHRASE");
+    const metadata = getConfig("METADATA");
+    const repeatedAdditionalDeposit = new U64(parseInt(getConfig("ADDITIONAL_DEPOSIT"), 10) || 0);
```

Comment:
I think an addtional deposit is not convenient to use.
Could you change it to the target deposit?

Reply:
Fixed